source/docs/software/commandbased/binding-commands-to-triggers.rst

```diff
@@ -1,240 +1,184 @@
 
-.. code-block:: java
+This binding cancels a command when a trigger changes from inactive to active (or, accordingly, when a button is initially pressed). the command is canceled on the iteration when the state changes, and will not be canceled again unless the trigger becomes inactive and then active again (or the button is released and re-pressed). Note that cancelling a command that is not currently running has no effect.
 
-   button.whenHeld(Command command)
+Binding a command to a joystick button
+--------------------------------------
 
-This binding schedules a command when a trigger changes from inactive to
-active (or, accordingly, when a button is initially pressed) and cancels
-it when the trigger becomes inactive again (or the button is released).
-The command will *not* be re-scheduled if it finishes while the trigger
-is still active.
+The most-common way to trigger a command is to bind a command to a button on a joystick or other HID (human interface device). To do this, users should use the ``JoystickButton`` class.
 
-whenInactive/whenReleased
+Creating a JoystickButton
 ~~~~~~~~~~~~~~~~~~~~~~~~~
 
-.. code-block:: java
+In order to create a ``JoystickButton``, we first need a Joystick.  All types of joysticks (including gamepads) are represented in code by the ``GenericHID`` class (`Java <https://first.wpi.edu/FRC/roborio/development/docs/java/edu/wpi/first/wpilibj/GenericHID.html>`__, `C++ <https://first.wpi.edu/FRC/roborio/development/docs/cpp/classfrc_1_1GenericHID.html>`__), or one of its subclasses:
 
-   trigger.whenInactive(Command command)
+.. tabs::
 
-.. code-block:: java
+  .. code-tab:: java
 
-   button.whenReleased(Command command)
+    Joystick exampleStick = new Joystick(1); // Creates a joystick on port 1
+    XboxController exampleController = new XboxController(2); // Creates an XboxController on port 2.
 
-This binding schedules a command when a trigger changes from active to
-inactive (or, accordingly, when a button is initially released). The
-command will be scheduled on the iteration when the state changes, and
-will not be re-scheduled unless the trigger becomes active and then
-inactive again (or the button is pressed and then re-released).
+  .. code-tab:: c++
 
-toggleWhenActive/toggleWhenPressed
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-.. code-block:: java
+    frc::Joystick exampleStick{1}; // Creates a joystick on port 1
+    frc::XBoxController exampleController{2} // Creates an XboxController on port 2
 
-   trigger.toggleWhenActive(Command command)
+.. note:: When creating a ``JoystickButton`` with an XboxController, it is usually a good idea to use the button enum (`Java <https://first.wpi.edu/FRC/roborio/development/docs/java/edu/wpi/first/wpilibj/XboxController.Button.html>`__, `C++ <https://first.wpi.edu/FRC/roborio/development/docs/cpp/classfrc_1_1XboxController.html#a1034633d4be2db9277b3864efa55b99b>`__) to get the button number corresponding to a given button.
```

Comment:
Is the button enum XboxController for 360 or XBox One? Those use different ways to interact with the system (DI vs XI) and therefore probably have different button IDs.

Reply:
360